features/netsocket/UDPSocket.cpp

```diff
@@ -33,6 +33,27 @@ nsapi_protocol_t UDPSocket::get_proto()
     return NSAPI_UDP;
 }
 
+int UDPSocket::modify_multicast_group(const SocketAddress &address, nsapi_socket_option_t socketopt)
+{
+    nsapi_ip_mreq_t mreq;
+
+    // Set up group address
+    mreq.imr_multiaddr = address.get_addr();
+    mreq.imr_interface = SocketAddress().get_addr();   // 0.0.0.0 INADDR_ANY
```

Comment:
`nsapi_addr_t()` would be simpler?

Comment is misleading - it's an "NSAPI_UNSPEC" address, rather than IPv4 0.0.0.0.